roles/ansible-test-splitter/files/list_changed_targets.py

```diff
@@ -199,38 +199,52 @@ def changed_files(self):
             ]
         return self.files
 
-    def modules(self):
-        """List the modules impacted by the change"""
+    def targets(self):
+        """List the test targets impacted by the change"""
         for d in self.changed_files():
-            if str(d).startswith("plugins/modules/"):
+            if str(d).startswith("tests/integration/targets/"):
+                # These are a special case, we only care that 'something' changed in that test
+                yield str(d).replace("tests/integration/targets/", "").split("/")[0]
+
+    def _path_matches(self, base_path):
+        # Simplest case, just a file name
+        for d in self.changed_files():
+            if str(d).startswith(base_path):
                 yield PosixPath(d)
 
+    def connection(self):
+        """List the connection plugins impacted by the change"""
+        yield from self._path_matches("plugins/connection/")
+
     def inventory(self):
         """List the inventory plugins impacted by the change"""
-        for d in self.changed_files():
-            if str(d).startswith("plugins/inventory/"):
-                yield PosixPath(d)
+        yield from self._path_matches("plugins/inventory/")
 
-    def module_utils(self):
-        """List the Python modules impacted by the change"""
+    def lookup(self):
+        """List the lookup plugins impacted by the change"""
+        yield from self._path_matches("plugins/lookup/")
+
+    def modules(self):
+        """List the modules impacted by the change"""
+        yield from self._path_matches("plugins/modules/")
+
+    def _util_matches(self, base_path, import_path):
```

Comment:
could be simplified using only parameter `import_path` as the util is always part of `plugins`

Reply:
The difference is that it also returns the ansible_collections.ABC.DEF.plugins.module_utils.XZY, which is then used to trigger tests against the dependencies of the utils.